### PR TITLE
Add time condition in scenes

### PR DESCRIPTION
**Pull request by Pierre-Gilles**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [ ] Did you add fake requests data for the demo mode (`front/src/config/demo.json`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Add a new condition in scenes:

Screenshot 2021-04-19 at 11 57 20

### CODE REVIEW

File: front/src/routes/scene/edit-scene/actions/CheckTime.jsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.friday" />
+      },
+      {
+        value: 'saturday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.saturday" />
+      },
+      {
+        value: 'sunday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.sunday" />
+      }
```
Comment (atrovato) [inline front/src/routes/scene/edit-scene/actions/CheckTime.jsx]: In `ScheduledTrigger.jsx` you
```
 const selectedWeekDaysOptions = this.props.trigger.days_of_the_week
      ? this.props.trigger.days_of_the_week.map(day => ({
          value: day,
          label: <Text id={`editScene.triggersCard.scheduledTrigger.daysOfTheWeek.${day}`} />
        }))
      : [];
```

Any reason to list it explicitly here?

**Review by atrovato — CHANGES_REQUESTED** [review]
On screen
image

The list is AM / PM format, but not the field, any way to fix it?
And blue "remove" cross on field is not really sexy.

For more, I tried it, but condition time is in GMT.
I setup a condition between 18:40 and 18h45, run at 18:40:47.
``` 
2021-04-20T18:40:47+0200 <debug> scene.actions.js:152 (Object.condition.check-time) Condition isAfterHour not verified. Now = Tue, 20 Apr 2021 16:40:47 GMT < afterDate = Tue, 20 Apr 2021 16:40:47 GMT
2021-04-20T18:40:47+0200 <error> scene.execute.js:20 () AbortScene [Error]: CONDITION_IS_AFTER_HOUR_NOT_VERIFIED
```

**Comment by Pierre-Gilles** [comment]
> The list is AM / PM format, but not the field, any way to fix it?

Right I forgot to add the user locale :) Fixing it.

> And blue "remove" cross on field is not really sexy.

I know it's terrible, I'll try to do something but it's the lib ^^

> I setup a condition between 18:40 and 18h45, run at 18:40:47.

Edit: Found the bug, fixing it now. I did a wrong usage of the dayjs library.

File: front/src/routes/scene/edit-scene/actions/CheckTime.jsx (hunk 1 above)
Comment (Pierre-Gilles) [reply front/src/routes/scene/edit-scene/actions/CheckTime.jsx]: No, in scheduled trigger it's listed explicity as well: 

Screenshot 2021-04-22 at 09 51 51

But you're right we could make a small array and .map, it'll be less verbose :)

**Comment by Pierre-Gilles** [comment]
We are lucky, the guy who build the react date picker lib just added the ability to customize the clear button 2 days ago! Nice! 

Screenshot 2021-04-22 at 10 22 58

**Comment by Pierre-Gilles** [comment]
So I put the cross in grey, it looks better. Still, if anyone is CSS master and want to make it even better i'm open :D 

Screenshot 2021-04-22 at 10 27 17

File: front/src/routes/scene/edit-scene/actions/CheckTime.jsx (hunk 1 above)
Comment (Pierre-Gilles) [reply front/src/routes/scene/edit-scene/actions/CheckTime.jsx]: Fixed

**Pull request merged**